### PR TITLE
Fix widget possible collisions

### PR DESCRIPTION
**Pull request by DvirMalka**
## Context

- Add sass compiler to the build process 
- Move everything under `.strigo-widget` to prevent breaking other websites' styles

## Tests

- [x] Testing locally
- [ ] Testing websites built on top of known CSS frameworks (bootstrap, buma, materialize, semantic, skeleton, pure)

## Screenshots

### CODE REVIEW

**Review by v0vs — COMMENTED** [review]
@DvirMalka ouch - important my eyes:). 
Maybe we can increase the CSS specificity by adding #strigo-widget id to the widget HTML element and target all styles starting with #strigo-widget?
It's gonna be tiresome to !important every CSS definition out there.

**Comment by DvirMalka** [comment]
@v0vs 
yes everything is already under `.strigo-widget` - but from my experience it's not enough... sometimes there are default behaviours to certain elements that we don't know about..

**Comment by DvirMalka** [comment]
will this method be enough to prevent, for instance - default styling of `i` `img` `button` or `p` ?

**Comment by DvirMalka** [comment]
hmm forget about the question - we can add our own styling under `.strigo-widget` or `#strigo-widget` and it'll be fine - incl. reset the original styles

**Comment by DvirMalka** [comment]
starwars-movealong

**Comment by DvirMalka** [comment]
@v0vs 
I added sass a compiler

**Comment by v0vs** [comment]
@DvirMalka Yeas, saas scoping sounds good. 
Also, I mean id selector has significantly higher specificity than class - https://specifishity.com/ 🙂, should help as well.

File: dist/development/strigo.sdk.js [hunk 1]
```diff
@@ -51,7 +51,7 @@
   }
   function generateCssURL(development, version) {
     if (development) {
-      return `http://localhost:${SDK_HOSTING_PORT}/styles/strigo.css`;
+      return `http://localhost:${"7005"}/styles/strigo.css`;
```
Comment (v0vs) [inline dist/development/strigo.sdk.js:54]: Port change crawled in... Let's exclude?

File: package-lock.json [hunk 2]
```diff
@@ -1,419 +1,57 @@
 {
   "name": "sdk",
   "version": "0.0.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
Comment (v0vs) [inline package-lock.json:4]: Looks like you used lower npm version than latest... Let's re-install using npm 8 (node 16)

File: package.json [hunk 3]
```diff
@@ -24,7 +24,8 @@
     "create-serve": "1.0.1",
     "dotenv": "16.0.0",
     "esbuild": "0.14.16",
-    "typescript": "4.6.3"
+    "typescript": "4.6.3",
+    "esbuild-sass-plugin": "^2.2.6"
```
Comment (v0vs) [inline package.json:28]: Let's make it a definitive version to avoid accidental upgrades.

**Review by v0vs — APPROVED** [review]

File: dist/development/strigo.sdk.js (hunk 1 above)
Comment (DvirMalka) [reply dist/development/strigo.sdk.js:54]: @nir0s 
If we exclude all `dist` via `.gitignore` - will the build be able to push the built versions back to git? If not - how would you suggest to do it (exclude locally without breaking the build) @v0vs  ?

**Comment by DvirMalka** [comment]
> @DvirMalka Yeas, saas scoping sounds good. Also, I mean id selector has significantly higher specificity than class - https://specifishity.com/ slightly_smiling_face, should help as well.

Changed main class to ID

File: dist/development/strigo.sdk.js (hunk 1 above)
Comment (DvirMalka) [reply dist/development/strigo.sdk.js:54]: @v0vs also, it doesn't really matter anyway - no one uses the `development` setup parameter other than the libro-demo setup. but I agree it's not Ideal. merging (it'll be overridden by the build process anyway)

**Pull request merged**